skrub/_drop_column_if_null.py

```diff
@@ -8,8 +8,23 @@
 
 
 class DropColumnIfNull(SingleColumnTransformer):
-    """Drop a single column if it contains only Null, NaN, or a mixture of null
-    values. If at least one non-null value is found, the column is kept."""
+    """Drop a single column if the fraction of Null or NaN values in the column
+    is larger than the given threshold.
+
+    By default, the threshold is set to `1.0`, so only columns that contain only
+    nulls or NaNs are dropped. Set the threshold to `None` to keep all columns.
+
+    Parameters
+    ----------
+    threshold : float, or None
+        Threshold of null values past which the column is dropped.
+    """
+
+    def __init__(self, threshold=1.0):
+        if threshold is not None:
```

Comment:
any checking of parameters should go in `fit` (in this case, at the start of `fit_transform`. the `__init__` of scikit-learn-compatible estimators never does anything else than `def __init__(self, **kwargs): self.__dict__.update(kwargs)`